docs/.readthedocs.yaml

```diff
@@ -14,7 +15,11 @@ build:
       - poetry config virtualenvs.create false
     post_install:
       # Install deps and build using poetry
-      - . "$READTHEDOCS_VIRTUALENV_PATH/bin/activate" && cd docs && poetry install
+      - . "$READTHEDOCS_VIRTUALENV_PATH/bin/activate" && cd docs && poetry install && cd ../icechunk-python && maturin develop && cd ../docs
```

Comment:
we can directly install from github with pip too if we know the commit ID. example:
```
git+https://github.com/earth-mover/icechunk.git@{self.full_commit}#subdirectory=icechunk-python
```

This will require maturin in the env, which we seem to have.